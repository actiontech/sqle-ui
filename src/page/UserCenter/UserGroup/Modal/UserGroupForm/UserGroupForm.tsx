import { Form, Input, Select, Switch } from 'antd';
import { useTranslation } from 'react-i18next';
import { UserGroupFormProps } from '.';
import EmptyBox from '../../../../../components/EmptyBox';
import IconTipsLabel from '../../../../../components/IconTipsLabel';
import { ModalFormLayout } from '../../../../../data/common';
import { nameRule } from '../../../../../utils/FormRule';

const UserGroupForm: React.FC<UserGroupFormProps> = (props) => {
  const { form, userList, isUpdate } = props;
  const { t } = useTranslation();

  return (
    <Form form={form} {...ModalFormLayout}>
      <Form.Item
        name="userGroupName"
        label={t('userGroup.userGroupField.userGroupName')}
        rules={[
          {
            required: true,
          },
          ...nameRule(119),
        ]}
      >
        <Input
          disabled={isUpdate}
          placeholder={t('common.form.placeholder.input')}
        />
      </Form.Item>
      <Form.Item
        name="userGroupDesc"
        label={t('userGroup.userGroupField.userGroupDesc')}
      >
        <Input.TextArea placeholder={t('common.form.placeholder.input')} />
      </Form.Item>
      <EmptyBox if={isUpdate}>
        <Form.Item
          name="isDisabled"
          label={
            <IconTipsLabel tips={t('userGroup.userGroupField.isDisabledTips')}>
              {t('userGroup.userGroupField.isDisabled')}
            </IconTipsLabel>
          }
          valuePropName="checked"
          initialValue={false}
        >
          <Switch />
        </Form.Item>
      </EmptyBox>
      <Form.Item
        name="userList"
        label={t('userGroup.userGroupField.userNameList')}
      >
        <Select
          showSearch
          mode="multiple"
          placeholder={t('common.form.placeholder.select')}
        >
          {userList?.map((user) => (
            <Select.Option key={user.user_name} value={user.user_name ?? ''}>
              {user.user_name}
            </Select.Option>
          ))}
        </Select>
      </Form.Item>
    </Form>
  );
};

export default UserGroupForm;
