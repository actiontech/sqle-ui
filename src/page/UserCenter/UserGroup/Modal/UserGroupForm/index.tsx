import { FormInstance } from 'antd';
import { IRoleTipResV1, IUserTipResV1 } from '../../../../../api/common';
import UserGroupForm from './UserGroupForm';

export type UserGroupFormProps = {
  form?: FormInstance<UserGroupFormField>;
  roleList?: IRoleTipResV1[];
  userList?: IUserTipResV1[];
  isUpdate?: boolean;
};
export type UserGroupFormField = {
  userGroupName: string;
  userGroupDesc?: string;
  roleList?: string[];
  userList?: string[];
  isDisabled?: boolean;
};

export default UserGroupForm;
