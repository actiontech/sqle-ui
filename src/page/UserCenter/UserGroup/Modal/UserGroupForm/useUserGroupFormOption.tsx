import { useEffect } from 'react';
import useRole from '../../../../../hooks/useRole';
import useUsername from '../../../../../hooks/useUsername';

const useUserGroupFormOption = (visible: boolean) => {
  const { roleList, updateRoleList } = useRole();
  const { usernameList, updateUsernameList } = useUsername();

  useEffect(() => {
    if (visible) {
      updateRoleList();
      updateUsernameList();
    }
  }, [updateRoleList, updateUsernameList, visible]);

  return {
    roleList,
    usernameList,
  };
};

export default useUserGroupFormOption;
