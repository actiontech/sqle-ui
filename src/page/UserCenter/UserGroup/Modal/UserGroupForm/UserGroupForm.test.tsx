import { render } from '@testing-library/react';
import UserGroupForm from '.';

describe('UserGroupForm', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <UserGroupForm
        roleList={[
          {
            role_name: 'role_name1',
          },
        ]}
        userList={[
          {
            user_name: 'user_name1',
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
