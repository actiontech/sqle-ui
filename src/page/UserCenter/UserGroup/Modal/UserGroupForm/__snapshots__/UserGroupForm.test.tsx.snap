// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGroupForm should match snapshot 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
      >
        <label
          class="ant-form-item-required"
          for="userGroupName"
          title="userGroup.userGroupField.userGroupName"
        >
          userGroup.userGroupField.userGroupName
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="userGroupName"
              placeholder="common.form.placeholder.input"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
      >
        <label
          class=""
          for="userGroupDesc"
          title="userGroup.userGroupField.userGroupDesc"
        >
          userGroup.userGroupField.userGroupDesc
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <textarea
              class="ant-input"
              id="userGroupDesc"
              placeholder="common.form.placeholder.input"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
      >
        <label
          class=""
          for="roleList"
          title="userGroup.userGroupField.roleNameList"
        >
          userGroup.userGroupField.roleNameList
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-multiple ant-select-show-search"
            >
              <div
                class="ant-select-selector"
              >
                <div
                  class="ant-select-selection-overflow"
                >
                  <div
                    class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                    style="opacity: 1;"
                  >
                    <div
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="roleList_list_0"
                        aria-autocomplete="list"
                        aria-controls="roleList_list"
                        aria-haspopup="listbox"
                        aria-owns="roleList_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="roleList"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                         
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  class="ant-select-selection-placeholder"
                >
                  common.form.placeholder.select
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-7"
      >
        <label
          class=""
          for="userList"
          title="userGroup.userGroupField.userNameList"
        >
          userGroup.userGroupField.userNameList
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-13"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-select ant-select-multiple ant-select-show-search"
            >
              <div
                class="ant-select-selector"
              >
                <div
                  class="ant-select-selection-overflow"
                >
                  <div
                    class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                    style="opacity: 1;"
                  >
                    <div
                      class="ant-select-selection-search"
                      style="width: 0px;"
                    >
                      <input
                        aria-activedescendant="userList_list_0"
                        aria-autocomplete="list"
                        aria-controls="userList_list"
                        aria-haspopup="listbox"
                        aria-owns="userList_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="userList"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                      <span
                        aria-hidden="true"
                        class="ant-select-selection-search-mirror"
                      >
                        
                         
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  class="ant-select-selection-placeholder"
                >
                  common.form.placeholder.select
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
